src/TagDependencyTrait.php

```diff
@@ -161,22 +186,21 @@ public static function loadModel(
      * Invalidate model tags.
      * @return bool
      */
-    public function invalidateTags()
+    public function invalidateTags($event)
     {
-
         /** @var TagDependencyTrait $this */
         \yii\caching\TagDependency::invalidate(
             $this->getTagDependencyCacheComponent(),
             [
                 static::commonTag(),
-                $this->objectTag()
+                $this->objectTag($event->name == ActiveRecord::EVENT_AFTER_UPDATE ? $event->changedAttributes : [])
```

Comment:
User can calls invalidate tags from own code. In this case user must creates an event. It's no best solution.